Add default translations for binding add-ons

wborn opened the pull request:
This makes the texts used by these add-ons translatable with Crowdin.

To keep the PR simple, it only adds default translations for add-ons which do not yet have any default translations properties file.
We can do follow up PRs for adding missing key/values to add-ons that already have these files or to remove duplications.

There are several add-ons in this PR that do have non-English translation files, so I'll upload those to Crowdin when the PR is merged.

Cossey commented:
Looks good! Can ok for the nzwateralerts, hpprinter and hccrubbishcollection bindings.

pacive reviewed (approved):
Good initiative! All good for smhi binding

hmerk reviewed (approved):
LGTM for Energenie and Wemo

t-8ch commented:
Ack for linuxinput

lolodomo commented:
> There are several add-ons in this PR that do have non-English translation files, so I'll upload those to Crowdin when the PR is merged.

This is the case for the netatmo binding as an example.
As the netatmo binding is being fully refactored, could you please exclude it from your list ? Because this is not time for translating it.
Note that for the meteoalerte binding, the texts in the default translation are in French. So either your exclude it until the binding is updated to provide English texts by default, or please take care to upload the default file as a French file in Crowdin.

lolodomo commented:
@wborn : how do you want to proceed ? Noone will check in details all these files. We are confident in your tool and we just merge if it builds ?

wborn commented:
> Noone will check in details all these files. We are confident in your tool and we just merge if it builds ?

I would keep it open for a couple of hours so there is some time for comments like yours and then merge it today.

bobadair reviewed (approved):
This is great! LGTM for the Alarm Decoder and Lutron bindings. Can you make the script available to sync new changes from the XML files to the default language properties files?

wborn commented:
It's not a script but a Maven plug-in 😉, see https://github.com/openhab/openhab-core/pull/2544.

You can run it using:

`mvn i18n:generate-default-translations`

ecdye reviewed (approved):
LGTM for OmniLink

mlobstein commented on bundles/org.openhab.binding.monopriceaudio/src/main/resources/OH-INF/i18n/monopriceaudio.properties:
```diff
@@ -0,0 +1,106 @@
+thing-type.monopriceaudio.amplifier.group.zone17.description = The Controls for Zone 17
+thing-type.monopriceaudio.amplifier.group.zone18.label = Zone 18
+thing-type.monopriceaudio.amplifier.group.zone18.description = The Controls for Zone 18
+thing-type.monopriceaudio.amplifier.group.zone2.label = Zone 2
```
It might be better if these are re-ordered numerically.

mlobstein commented on bundles/org.openhab.binding.nuvo/src/main/resources/OH-INF/i18n/nuvo.properties:
```diff
@@ -0,0 +1,142 @@
+thing-type.nuvo.amplifier.group.zone18.description = The Controls for Zone 18
+thing-type.nuvo.amplifier.group.zone19.label = Zone 19
+thing-type.nuvo.amplifier.group.zone19.description = The Controls for Zone 19
+thing-type.nuvo.amplifier.group.zone2.label = Zone 2
```
It might be better if these are re-ordered numerically.

mlobstein commented on bundles/org.openhab.binding.atlona/src/main/resources/OH-INF/i18n/atlona.properties:
```diff
@@ -0,0 +1,236 @@
+thing-type.atlona.pro3-1616m.group.port18.description = Output Port 18 Channels
+thing-type.atlona.pro3-1616m.group.port19.label = Port 19
+thing-type.atlona.pro3-1616m.group.port19.description = Output Port 19 Channels
+thing-type.atlona.pro3-1616m.group.port2.label = Port 2
```
Another that should be ordered numerically.

mlobstein commented on bundles/org.openhab.binding.autelis/src/main/resources/OH-INF/i18n/autelis.properties:
```diff
@@ -0,0 +1,369 @@
+thing-type.autelis.jandy.channel.equipment-aux18.description = The current state of auxiliary 18
+thing-type.autelis.jandy.channel.equipment-aux19.label = Auxiliary 19
+thing-type.autelis.jandy.channel.equipment-aux19.description = The current state of auxiliary 19
+thing-type.autelis.jandy.channel.equipment-aux2.label = Auxiliary 2
```
Another that should be ordered numerically.

mlobstein commented on bundles/org.openhab.binding.daikin/src/main/resources/OH-INF/i18n/daikin.properties:
```diff
@@ -0,0 +1,149 @@
+channel-type.daikin.acunit-energyheatingcurrentyear-11.description = The energy usage for heating this year November
+channel-type.daikin.acunit-energyheatingcurrentyear-12.label = Energy Heating Current Year December
+channel-type.daikin.acunit-energyheatingcurrentyear-12.description = The energy usage for heating this year December
+channel-type.daikin.acunit-energyheatingcurrentyear-2.label = Energy Heating Current Year February
```
Another that should be ordered numerically.

kabili207 reviewed (approved):
LIRC looks good

lolodomo commented:
Time to inject all that in Crowdin :)

lolodomo commented:
@wborn : do you have an easy way to rerun your command for the `same list of bindings`  ?
I already know that some entries are missing for few bindings because I/we recently merged PRs that did not care about this default properties file, like wled or omnikonverter for example.

wborn commented:
On the CLI you can do something like this with Bash at the repo root:

```
$ mvn i18n:generate-default-translations
$ git diff-tree --no-commit-id --name-only -r 07a1976c817e92d0b1274b4686cabd3442c8ec12 | xargs -I{} git add {}
$ git commit -m "Modified translations"
$ git add .
$ git reset --hard
$ git reset HEAD^
$ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   bundles/org.openhab.binding.atlona/src/main/resources/OH-INF/i18n/atlona.properties
	modified:   bundles/org.openhab.binding.autelis/src/main/resources/OH-INF/i18n/autelis.properties
	modified:   bundles/org.openhab.binding.benqprojector/src/main/resources/OH-INF/i18n/benqprojector.properties
	modified:   bundles/org.openhab.binding.daikin/src/main/resources/OH-INF/i18n/daikin.properties
	modified:   bundles/org.openhab.binding.deutschebahn/src/main/resources/OH-INF/i18n/deutschebahn.properties
	modified:   bundles/org.openhab.binding.hydrawise/src/main/resources/OH-INF/i18n/hydrawise.properties
	modified:   bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/i18n/magentatv.properties
	modified:   bundles/org.openhab.binding.monopriceaudio/src/main/resources/OH-INF/i18n/monopriceaudio.properties
	modified:   bundles/org.openhab.binding.nuvo/src/main/resources/OH-INF/i18n/nuvo.properties
	modified:   bundles/org.openhab.binding.omnikinverter/src/main/resources/OH-INF/i18n/omnikinverter.properties
	modified:   bundles/org.openhab.binding.teleinfo/src/main/resources/OH-INF/i18n/teleinfo.properties
	modified:   bundles/org.openhab.binding.wled/src/main/resources/OH-INF/i18n/wled.properties
```

It reruns the plugin on all add-ons and then only adds the files in my commit to another commit... and then cleans all modified/untracked files in the workspace and undos the commit.
